apps/web/pages/api/trpc/[trpc].ts

```diff
@@ -30,34 +30,64 @@ export default trpcNext.createNextApiHandler({
    * @link https://trpc.io/docs/caching#api-response-caching
    */
   responseMeta({ ctx, paths, type, errors }) {
```

Comment:
@zomars @PeerRich I've reorganized this function a bit to prevent all logic from running each time the function is called. Instead I've split the large if statement into multiple smaller statements that hopefully make the code a bit easier to follow.

Additionally, the previous logic was to completely exclude a given path from being cached. I've updated the logic to instead of able to respond with different levels of caching depending on the value of `paths`.